src/btllib/aahash.cpp

```diff
@@ -0,0 +1,77 @@
+#include "btllib/aahash.hpp"
+#include "btllib/aahash_consts.hpp"
+#include "btllib/nthash_lowlevel.hpp"
+
+namespace btllib {
+std::vector<SpacedSeed>
+aa_parse_seeds(const std::vector<std::string>& seeds)
+{
+  std::vector<SpacedSeed> seed_vec;
+  // convert vector of string to vector of vector of unsigned
+  for (const auto& seed : seeds) {
+    SpacedSeed seed_vec_tmp;
+    for (const auto& c : seed) {
+      // push back the unsigned value of the char by type conversion
+      seed_vec_tmp.push_back((unsigned)(c - '0'));
```

Comment:
If the seed string contains an out-of-bounds value for the level, it should be caught here. Users may run into ambiguous errors otherwise.

Reply:
I added the check but I think it is more appropriate to do it in another place.